### DISCUSSION TITLE
Get lyrics time

### DISCUSSION THREAD

**Discussion by Nelfstor** [melanchall/drywetmidi#293] [Q&A] [answered]
Hello! 

Thanks for your library! i'm using it a while.

I'm trying to get lyrics time and attach them to the notes. 

With notes everything is fine - i use converter and get times for them which are adequate and checked. 

With lyrics first i'm getting absolute time by adding the deltaTimes, but numbers are significantly different. And actually i don't understand what they are. {0, 12, 24, 6, 300}

( probably "300" - is an event came from another track?)

In comments near deltaTimes it is written: 
        /// Delta-time is in some fraction of a beat (or a second, for recording a track with SMPTE times), as specified
        /// by the file's time division.

But timeDivision of tempoMap doesn't hold the value. 

In documentation i haven't found the right conversion for this. [https://melanchall.github.io/drywetmidi/articles/high-level-managing/Time-and-length.html](here)

Could you help, please?

**Comment by melanchall** [comment, marked as answer]
Hi,

> ( probably "300" - is an event came from another track?)

You didn't provide an algorithm or even code, so I can't answer the question. I'm not a mind reader unfortunately :-)

> i'm getting absolute time by adding the deltaTimes

The main goal of the library is to provide easy solutions for common practical tasks. Of course you don't need to sum delta-times. Here a sample code which, I hope, answers your questions.

```csharp
var tempoMap = midiFile.GetTempoMap();
var lyricTimedEvents = midiFile
    .GetTimedEvents()
    .Where(e => e.Event.EventType == MidiEventType.Lyric);

foreach (var e in lyricTimedEvents)
{
    var metricTime = e.TimeAs<MetricTimeSpan>(tempoMap);
    var lyricText = ((LyricEvent)e.Event).Text;
}
```

Please read articles in the **Articles** section of the [library docs](https://melanchall.github.io/drywetmidi).

**Reply by Nelfstor** [reply]
Hello, Maxim!

Thanks a lot for the quick response!

Before that i've received lyrics in other way! Here how it was:
```
var allChanks = midiFile.GetTrackChunks();
            var lyricEvents = new List<LyricEvent>();
            foreach (var chank in allChanks)
            {
                using (var objectManager = chank.ManageTimedEvents())
                {
                    lyricEvents.AddRange(objectManager.Objects
                                                   .Where(e => e.Event is LyricEvent)
                                                   .Select(e => (LyricEvent)e.Event)
                                                   .ToList());
                }
            }
```
so i couldn't use that lyricEvent as TimedEvent!

Now i used your code and get the result! 

Thank a lot! 
Благодарю!
